### PR TITLE
fix(module:message): fix container instance not destroyed in HMR

### LINKED ISSUE #3845: Message: 在 HMR 开发模式下，会导致 Attempt to use a destroyed 异常
## 重现

https://stackblitz.com/edit/angular-uif43a?file=src%2Fapp%2Fapp.component.ts

**步骤：**

- 打开 Stackblitz 后，在 `template: ''` 录入任意字符，触发一次 HMR 的加载
- 点击按钮抛出 `Attempt to use a destroyed`

虽然在生产环境下无影响，但对于 HMR 开发环境下，会非常麻烦，特别是在需要触发 Message 时该错误会导致后续的动作无法继续，非要刷新 `Command + R` 后才会生效。

Version:

- ng-zorro-antd latest

### PR DESCRIPTION
**Pull request by wendzhue**
close #3845

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: #3845 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by wendzhue** [comment]
@cipchk You can try to build from this PR and try it with your demo. It should work now.

**Comment by cipchk** [comment]
@wendzhue Yes, already working in HMR mode (includes other projects).

File: components/core/services/nz-singleton.service.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+ */
+
+// tslint:disable no-any
+
+import { Injectable } from '@angular/core';
+
+import { environment } from '../environments/environment';
+
+interface SingletonRegistryItem {
+  refCount: number;
```
Comment (hsuanxyz) [inline components/core/services/nz-singleton.service.ts]: `refCount` 没有使用到

Comment (wendzhue) [reply components/core/services/nz-singleton.service.ts]: 嗯，移除掉

**Pull request merged**